drift-transport-netty/src/main/java/io/airlift/drift/transport/netty/codec/HeaderTransport.java

```diff
@@ -240,4 +244,38 @@ private static int readVariableLengthInt(ByteBuf messageHeader)
 
         return result;
     }
+
+    public static Optional<FrameInfo> tryDecodeFrameInfo(ByteBuf input)
+    {
+        ByteBuf buffer = input.retainedDuplicate();
+        try {
+            if (buffer.readableBytes() < FRAME_HEADER_SIZE) {
+                return Optional.empty();
+            }
+            // skip magic
+            buffer.readShort();
+            short flags = buffer.readShort();
+            boolean outOfOrderResponse = isOutOrOrderResponseSupported(flags);
+            int sequenceId = buffer.readInt();
+            int headerSize = buffer.readShort() << 2;
+
+            if (buffer.readableBytes() < Byte.BYTES) {
+                return Optional.empty();
+            }
+            byte protocolId = buffer.getByte(buffer.readerIndex());
+            Protocol protocol = Protocol.getProtocolByHeaderTransportId(protocolId);
+
+            if (buffer.readableBytes() < headerSize) {
+                return Optional.empty();
+            }
+            buffer.skipBytes(headerSize);
+            SimpleFrameInfoDecoder simpleFrameInfoDecoder = new SimpleFrameInfoDecoder(HEADER, protocol, outOfOrderResponse);
+            return simpleFrameInfoDecoder.tryDecodeFrameInfo(buffer)
+                    // prefer sequence id from the frame header
```

Comment:
Why would they be different? Should we fail if they are?

Reply:
No idea. You are right, let me add an assertion.